### PR TITLE
Huge refactor and add Scaleway+Docker+Swarm

### PR DESCRIPTION
**Pull request by hazcod**

### CODE REVIEW

File: modules/node/security-groups.tf [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+    for_each = local.udpPorts
+    content {
+      action    = "accept"
+      port      = inbound_rule.value
+      protocol  = "UDP"
+    }
+  }
+}
+
+resource "scaleway_security_group" "swarm_manager" {
```
Comment (kindermoumoute) [inline modules/node/security-groups.tf]: The resource was renamed to `scaleway_instance_security_group`

File: modules/node/variables.tf [hunk 2]
```diff
@@ -13,7 +13,7 @@ variable "ssh_root_public_key" {
 variable "ssh_tech_public_key" {
 }
 
-variable "region" {
+variable "scaleway_region" {
   default = "ams1"
```
Comment (kindermoumoute) [inline modules/node/variables.tf]: region is now called `nl-ams`, but you must provide a zone which is  `nl-ams-1`. See the new doc about it  ==> https://www.terraform.io/docs/providers/scaleway/guides/regions_and_zones.html

File: modules/node/security-groups.tf [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+  name        = "swarm_managers"
+  description = "Allow SSH, HTTP(S) and internal Swarm traffic"
+
+  inbound_default_policy  = "drop"
+  outbound_default_policy = "accept"
+  stateful                = true
```
Comment (kindermoumoute) [inline modules/node/security-groups.tf]: you don't need to pass stateful anymore (it's stateful by default)

**Review by kindermoumoute — CHANGES_REQUESTED** [review]
Few comments that may explain you errors

File: modules/node/security-groups.tf (hunk 3 above)
Comment (hazcod) [reply modules/node/security-groups.tf]: Thanks, will keep it for clarity tough.

File: modules/node/variables.tf (hunk 2 above)
Comment (hazcod) [reply modules/node/variables.tf]: FIxed, thanks!

File: modules/node/security-groups.tf (hunk 1 above)
Comment (hazcod) [reply modules/node/security-groups.tf]: Aha, thanks!

File: modules/node/security-groups.tf (hunk 3 above)
Comment (kindermoumoute) [reply modules/node/security-groups.tf]: The field does not exist anymore in the new resource, terraform might throw an error.

Comment (hazcod) [reply modules/node/security-groups.tf]: Fixed in latest commit.

**Review by kindermoumoute — APPROVED** [review]
LGTM for the terraform changes, if you're having any issue with Scaleway provider the quickest way to get help is through a [GitHub issue](https://github.com/terraform-providers/terraform-provider-scaleway/issues) or on [Scaleway community Slack](https://slack.scaleway.com/)

File: modules/node/nodes.tf [hunk 4]
```diff
@@ -0,0 +1,55 @@
+# public ip address for scaleway instance
+resource "scaleway_ip" "node_ip" {
+  count = "${var.instance_count}"
+}
+
+# scaleway instance
+resource "scaleway_server" "instance" {
```
Comment (kindermoumoute) [inline modules/node/nodes.tf]: You may want to use the new resource [`scaleway_instance_server`](https://www.terraform.io/docs/providers/scaleway/r/instance_server.html) which comes with [the placement group feature](https://www.terraform.io/docs/providers/scaleway/r/instance_placement_group.html)

File: modules/node/nodes.tf [hunk 5]
```diff
@@ -4,15 +4,18 @@ resource "scaleway_ip" "node_ip" {
...
 # scaleway instance
-resource "scaleway_server" "instance" {
+resource "scaleway_instance_server" "instance" {
 
+  zone           = "${var.scaleway_zone}"
   count          = "${var.instance_count}"
   name           = "${terraform.workspace}-manager-${count.index + 1}"
   image          = "${data.scaleway_image.docker.id}"
   type           = "${var.manager_instance_type}"
   security_group = "${scaleway_instance_security_group.node_rules.id}"
```
Comment (kindermoumoute) [inline modules/node/nodes.tf]: `security_group_id`

File: modules/node/nodes.tf [hunk 6]
```diff
@@ -4,15 +4,18 @@ resource "scaleway_ip" "node_ip" {
...
   public_ip      = "${element(scaleway_ip.swarm_manager_ip.*.ip, count.index)}"
   tags           = var.tags
+  policy_type    = "max_availability"
```
Comment (kindermoumoute) [inline modules/node/nodes.tf]: `placement_group_id`, the placement group policy must be set in a `scaleway_instance_placement_group` resource

**Review by kindermoumoute — CHANGES_REQUESTED** [review]

**Pull request closed without merging**